### ISSUE TITLE
Text Truncation Ellipsis

### ISSUE THREAD

**Issue by AlbinoGeek** [fyne-io/fyne#1659]
TextProvider should offer **optional** text truncation ellipsis along with its wrapping features.

Why optional? Some Material Design components specifically dictate not truncating text.

Related: #332

**Comment by andydotxyz** [comment]
Indeed. We envisaged that this would be added to the list of truncate/wrap types - just did not have time to implement it when adding truncation.

**Comment by fgm** [comment]
One thing not to forget about ellipsizing is supporting middle truncation instead of just extremities. It's quite often the most useful way to do it. See e.g. Pango for the API: https://lazka.github.io/pgi-docs/Pango-1.0/classes/Layout.html#Pango.Layout.set_ellipsize

**Comment by andydotxyz** [comment]
Supporting the middle break (correctly) will require integrating the ongoing work with [go-text](/go-text) project, so maybe the first version will be just word ending, and we could build on it from there....

**Comment by andydotxyz** [comment]
Could do trailing first as it is the most common and easier than other combinations.

**Comment by oxplot** [comment]
I'd like to add that middle/beginning ellipsis are useful for file paths as the filename is often the most important part.
